### ISSUE TITLE
Inserting blocks with the keyboard alone

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#834]
Splitting this out from previous tickets where it was related to the inserter.

This feature is nice to have, but it is not critical for the flow to work. It is low priority and should be tackled if we can find a nice and clean way to build it, and only after more important pieces are in place. See also #798.

## Behavior

When the caret is on a new line, in an empty block, you can either start typing, or you can use one of these commands to do cool things:

- Type `- ` to start a list, other Markdown tricks work too.
- Type `# ` for headlines. Note the space.
- Type `> ` for quotes

![newline](https://cloud.githubusercontent.com/assets/1204802/26201173/026af178-3bd2-11e7-92d1-70d361474cd7.png)

If If you type `/`, you invoke the inline inserter and set focus on its search field:

![inserter invoked by](https://cloud.githubusercontent.com/assets/1204802/26201182/0c670720-3bd2-11e7-92fa-214003c46f65.png)

Type out the blockname and press enter to insert:

![inserter searched](https://cloud.githubusercontent.com/assets/1204802/26201196/1e9c4d56-3bd2-11e7-85a2-70d5308824e9.png)

**Comment by nylen** [comment]
> When the caret is on a new line, in an empty block

In the mockups it looks like the caret is inside an empty paragraph inside an existing (larger, non-empty) text block.

**Comment by jasmussen** [comment]
> In the mockups it looks like the caret is inside an empty paragraph inside an existing (larger, non-empty) text block.

Yeah, in the mockup the blocks have already faded out :trollface: 

That being said, I do follow you, and it speaks to a conversation we had just now about perhaps showing the boundaries slightly more often for text, or even having them be faintly visible all the time. Something I'll think about more.

**Comment by jasmussen** [comment]
Just cross referencing with #2284 so this ticket can be closed when that gets merged in.

**Comment by mtias** [comment]
Done in #2630.
